datafusion/common/src/scalar.rs

```diff
@@ -3468,22 +3541,30 @@ mod tests {
         }
 
         // decimal scalar to array
-        let array = decimal_value.to_array();
+        let array = decimal_value
+            .to_array()
+            .expect("Failed to convert to array");
```

Comment:
I think it makes sense to use `.expect` here so that we can have more information on test failure. See my other comment https://github.com/apache/arrow-datafusion/pull/7901#issuecomment-1780426607